### ISSUE TITLE
SQL query contains the percent sign that should remain, for example in a LIKE SQL query

### ISSUE THREAD

**Issue by henrikek** [microsoft/mssql-django#369]
https://github.com/microsoft/mssql-django/commit/adc01a58af835f0ba08c69947b081cea0b6049d1#commitcomment-140173759

https://github.com/microsoft/mssql-django/pull/354

**Comment by dauinsight** [comment]
Hi @henrikek,

This makes sense to me, but you could also provide us with a test case to expedite our investigation process. Thank you.

**Comment by henrikek** [comment]
I have tried to reproduce this with a normal django query but I have not succeeded. I saw this problem before in combination with Django rest framework and SearchFilter, but I'm sorry but I haven't been able to reproduce this, so maybe it's not a problem. But at the same time, I know that if you format a string in two parts as the new code looks like, problems with the percent sign can cause it.

**Comment by sparrowt** [comment]
In a virtualenv for this project, running `python manage.py shell` and then the following is sufficient to reproduce the issue:
```
from django.db.models import F, Min
from testapp import models
models.Number.objects.values('integer').annotate(myannotation=Min(F('integer') % 3))
```

This is somewhat contrived, but we hit this bug in real code and it is blocking us from upgrading to `mssql-django==1.4.2`. In the mean time we have had to manually patch in the fix for https://github.com/advisories/GHSA-vmqv-47j8-gwv8 (CVE-2024-26164) on top of 1.3.

**Comment by dauinsight** [comment]
Thanks for your responses.

It appears the bug stems from performing an additional string formatting operation using %. Doing so removes the extra % used for escaping the modulo operator. 

Example:

'%% %s' -> '% (value)'  After the first string format.
'% (value)' -> Error       After the second string format. Here python expects to insert a parameter but it can't find it.

**Comment by sparrowt** [comment]
Can this be closed now that #378 is merged?
